### DISCUSSION TITLE
How to deprioritize exact matches

### DISCUSSION THREAD

**Discussion by chrisgrieser** [hrsh7th/nvim-cmp#1393] [Q&A] [answered]
in this case, I have a snippet and an LSP suggestion. In my sources, snippets are higher ranked, and the snippet gets ranked higher (image at the top). But as soon as there is an exact match, that exact match gets prioritized (image at the bottom).

Is there any way to change that behavior, i.e. deprioritize exact matches? I tried increasing source priority or priority-weight, but it seems that exact matches always get the ranked at the top regardless.

<img width="776" alt="Pasted image 2023-01-05 02 12 42" src="https://user-images.githubusercontent.com/73286100/210679412-f6969bef-1d3d-4966-b20f-aa378f42f321.png">

<img width="674" alt="Pasted image 2023-01-05 02 11 21" src="https://user-images.githubusercontent.com/73286100/210679281-66bfcb44-41a3-4aab-b15e-1c368aa06cf5.png">

**Comment by chrisgrieser** [comment, marked as answer]
okay, I actually found it myself by searching the source code for "exact" 😝 

for anyone finding this via search: in the `cmp.setup()` call, you have to modify the list of `comparators`. I just copied the list but left out `compare.exact`, which disabled prioritizing exact matches :)
https://github.com/hrsh7th/nvim-cmp/blob/main/lua/cmp/config/default.lua#L59-L69

**Reply by doulighan** [reply]
Does this completely disable exact matches, or just de-prioritize? I'm still getting them, finding exact matches to be useless with copilot

**Reply by chrisgrieser** [reply]
if you leave out `compare.exact`, it disregards exact matches when ordering the results. If you just move it down in the list of comparators, the importance of exact matches is far lowered. In both cases, exact matches are still listed in suggestions.

I would just try it out with which setting you get results you are more content with

**Reply by Sam-programs** [reply]
You could also make a custom comparator that is the opposite of `compare.exact`.
```lua
function(entry1,entry2)
  if(entry1.exact ~= entry2.exact) then
     return not entry1.exact
  end
  return nil
end
```
That way exact matches always go to the bottom of the list.
